compiler/rustc_lint/src/lints.rs

```diff
@@ -0,0 +1,51 @@
+use rustc_ast::{LitIntType, LitKind};
+use rustc_errors::fluent;
+use rustc_macros::LintDiagnostic;
+use rustc_span::{Span, Symbol};
+
+// FIXME: `#[derive(LintDiagnostic)] with #[suggestion_*]` does not fully output the suggestion,
+// bug maybe?
+// expected: ^^^^^^ help: use an inclusive range instead: `<suggestion>`
+// actual  : ^^^^^^ help: use an inclusive range instead
+//
+// #[derive(LintDiagnostic)]
+// #[diag(lint::range_endpoint_out_of_range)]
+pub struct RangeEndpointOutOfRange<'a> {
+    pub ty: &'a str,
+    // #[suggestion_short(code = "{start}..={literal}{suffix}", applicability = "machine-applicable")]
```

Comment:
```suggestion
    // #[suggestion_short(code = "{start}..={literal}{suffix}", applicability = "machine-applicable")]
```

If using `suggestion`, as in your manual implementation below, works, then just use `#[suggestion(..)]` here rather than `#[suggestion_short]`?